shuffle/cli/src/node.rs

```diff
@@ -1,31 +1,105 @@
-        println!("Creating node config in {}", shuffle_dir.display());
-        fs::create_dir_all(&shuffle_dir)?;
+use diem_config::config::NodeConfig;
+use diem_types::{chain_id::ChainId, on_chain_config::VMPublishingOption};
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+pub fn handle(genesis: Option<String>) -> Result<()> {
+    let home = Home::new(shared::get_home_path().as_path())?;
+    if !home.get_shuffle_path().is_dir() {
+        println!(
+            "Creating node config in {}",
+            home.get_shuffle_path().display()
+        );
+
+        create_node(&home, genesis)
     } else {
-        println!("Accessing node config in {}", shuffle_dir.display());
+        println!(
+            "Accessing node config in {}",
+            home.get_shuffle_path().display()
+        );
+        if genesis.is_some() {
+            return Err(
+                anyhow::anyhow!(
+                    "Unable to set genesis on an already created node. rm -rf ~/.shuffle to recreate node; you will lose state"
+                )
+            );
+        }
+
+        start_node(&home)
     }
-    let publishing_option = VMPublishingOption::open();
+}
 
-    // prepare a deterministic generator so that recreated test environments
-    // have the same waypoints, configurations, etc
-    let rng = rand::rngs::StdRng::seed_from_u64(0);
+fn create_node(home: &Home, genesis: Option<String>) -> Result<()> {
+    fs::create_dir_all(home.get_shuffle_path())?;
+
+    let publishing_option = VMPublishingOption::open();
+    let genesis_modules = genesis_modules_from_path(&genesis)?;
     diem_node::load_test_environment(
-        Some(shuffle_dir.join("nodeconfig")),
+        Some(PathBuf::from(home.get_node_config_path())),
         false,
         Some(publishing_option),
-        diem_framework_releases::current_module_blobs().to_vec(),
-        rng,
+        genesis_modules,
+        rand::rngs::OsRng,
+    );
+    Ok(())
+}
+
+fn start_node(home: &Home) -> Result<()> {
+    println!("\tLog file: {:?}", home.get_validator_log_path());
+    println!("\tConfig path: {:?}", home.get_validator_config_path());
+    println!("\tDiem root key path: {:?}", home.get_root_key_path());
+    println!("\tWaypoint: {}", home.read_genesis_waypoint()?);
+    // Configure json rpc to bind on 0.0.0.0
+    let mut config = NodeConfig::load(home.get_validator_config_path()).unwrap();
+    config.json_rpc.address = format!("0.0.0.0:{}", config.json_rpc.address.port())
+        .parse()
+        .unwrap();
+    println!("\tJSON-RPC endpoint: {}", config.json_rpc.address);
+    config.json_rpc.stream_rpc.enabled = true;
+    println!("\tStream-RPC enabled!");
+    config.api.enabled = true;
+    println!("\tREST API enabled!");
+    println!("\tREST API: {}", config.api.address);
+
+    println!(
+        "\tFullNode network: {}",
+        config.full_node_networks[0].listen_address
     );
+    println!("\tChainId: {}", ChainId::test());
```

Comment:
may get the chain id from `config`, otherwise, is it possible a user change it in the configuration and won't be printed here.

Reply:
It's surprisingly hard to get the chain id from the existing `config`. It only exists on the genesis_storage.json which is only used in the genesis ceremony on node creation. Reading it introduces a strange chicken egg scenario where 
1. I need to read the chain id from .move storage before starting the node
2. I need to start the node before I can read the onchain storage.

Punting for now but will revisit.